tests/test_examples.py

```diff
@@ -1534,7 +1542,11 @@ def test_frozen_environment(tmp_path, request):
             assert frozen_file.exists()
             actual_frozen_paths.add(frozen_file)
 
-        assert expected_frozen_paths == actual_frozen_paths, (
-            f"Expected: {sorted(str(p) for p in expected_frozen_paths)}\n"
+        assert set(expected_frozen_paths.values()) == actual_frozen_paths, (
+            f"Expected: {sorted(str(p) for p in expected_frozen_paths.values())}\n"
             f"Found: {sorted(str(p) for p in actual_frozen_paths)}"
         )
+
+        for method, path in expected_frozen_paths.items():
+            actual_config = json.load(open(path))
```

Comment:
```suggestion
            actual_config = json.loads(path.read_text())
```

`open` outside of a context manager doesn't close the file handle until garbage collection. File handles should be closed when files are no longer needed to avoid resource leaks.

Reply:
You may also want to use a different variable name than `path` because it is too similar to the `Path` object.